### ISSUE TITLE
Bug when commenting macros (# is causing issues)

### ISSUE THREAD

**Issue by klonq123** [christophschlosser/doxdocgen#142]
# Description

There is a small problem when documenting a macro.

- The < #de > is removed from the macro definition
- No line break is inserted after the comment

See code example below and screenshots attached

![doxygen_bug](https://user-images.githubusercontent.com/23451157/69502457-d45cd080-0f0f-11ea-94ba-b479055b7c8b.png)
![doxygen_bug_result](https://user-images.githubusercontent.com/23451157/69502458-d45cd080-0f0f-11ea-8ec8-4898258b345a.png)

## Code example

```Cpp
/** <press enter>
#define MY_MACRO 1
```

### Expected result

```Cpp
/**
 * @brief 
 * 
 */
#define MY_MACRO 1
```

### Actual result

```Cpp
/**
 * @brief 
 * 
 */fine MY_MACRO 1
```

**Comment by christophschlosser** [comment]
![bug-142](https://user-images.githubusercontent.com/2466365/69574250-fa45ac00-0fc7-11ea-9e45-0edfe45e90cf.gif)

Could you please help me out a bit here?

Your system specs and versions used would help a lot.

Can you reproduce it in an empty file or is there any other code around it?

**Comment by klonq123** [comment]
It seems only one particular file is affected other files in the same project don't seem to be affected.
Attached for your viewing pleasure
C:\Users\Mathieu\source\pvconnex\lib\gsm\include\Telit_GE910_Quad_V3.h

**Comment by klonq123** [comment]
Sorry, I can't attach header files.
Here is a WeTransfer link : https://we.tl/t-apHSNmAx4T

**Comment by christophschlosser** [comment]
Thanks for your feedback.
Could you copy and paste the code in here please?

**Comment by klonq123** [comment]
`
#pragma once

#include <mbed.h>
#include "Logger.h"
#include "AT_Commands.h"

/**
 * @brief The interval at which to process the queue
 */
#define TELIT_TASK_INTERVAL 10.0f

/**
 * @brief The system is set to use verbose reporting
 */
#define SYSTEM_CMEE_LEVEL CMEE_VERBOSE

/**
 * @brief Command interface style
 */
#define SYSTEM_SELINT_SETTING SELINT_2

/**
 * @brief The system uses the enhanced operation mode
 */
#define SYSTEM_REGMODE_SETTING REGMODE_1_ENABLED

/**
 * @brief The system uses the improved SMS mode
 */
#define SYSTEM_SMSMODE_SETTING SMSMODE_1_ENABLED

/**
 * @brief The return code used for failed operations. 
 */
#define TELIT_FAILURE -1

/**
 * @brief The time needed for the GSM to start and synchronise with the network
 * This parameter will be used when turning on the device or unlocking the device with PIN
 */
#define SYSTEM_STARTUP_TIME 5000

/**
 * @brief Timeout for typical AT comand
 */
#define AT_TIMEOUT 8000

/**
  * @brief Timeout for sending SMS 
  */
#define SEND_SMS_TIMEOUT 30000

/**
 * @brief The time in milliseconds to wait between two successive commands
 */
#define TIME_BETWEEN_COMMANDS 20

/**
  * @brief The buffer size for all transactions
  */
#define AT_BUFFER_SIZE 256

/**
 * @brief Select interface setting of the device.
 * Provides compatibility between different products.
 * The GE910-Quad-V3 is by default in mode #2
 */
typedef enum SELINT_Type
{
    SELINT_FAILED = -1,
    SELINT_0 = 0,
    SELINT_1 = 1,
    SELINT_2 = 2
} SELINT_Type;

/**
 * @brief Registration operation mode.
 * REGMODE=1 provides the best compatibility with ETSI specifications.
 * This is the default setting of GE910-Quad-V3
 */
typedef enum REGMODE_Type
{
    REGMODE_FAILED = -1,
    REGMODE_0_DISABLED = 0,
    REGMODE_1_ENABLED = 1
} REGMODE_Type;

/**
 * @brief SMS commands operation mode
 * SMSMODE=1 is the improved handling of SMS operations
 * This is the default ssetting of GE910-Quad-V3
 */
typedef enum SMSMODE_Type
{
    SMSMODE_FAILED = -1,
    SMSMODE_0_DISBALED = 0,
    SMSMODE_1_ENABLED = 1,
    SMSMODE_2_CHECK_SERVICE_CENTER = 2
} SMSMODE_Type;

typedef struct PING_Resp
{
    bool success;
    uint8_t retry;
    char *ipAddress;
    uint16_t replyTime;
    uint16_t ttl;
} PING_Resp;

typedef enum SocketIdentifier
{
    TELIT_SOCKET_1 = 1,
    TELIT_SOCKET_2 = 2,
    TELIT_SOCKET_3 = 3,
    TELIT_SOCKET_4 = 4,
    TELIT_SOCKET_5 = 5,
    TELIT_SOCKET_6 = 6
} SocketIdentifier;

typedef struct SCFG_Resp
{
    SocketIdentifier socketIdentifier;
    PDP_CONTEXT_ID cid;
    uint16_t packetSize;
    uint16_t exchangeTimeout;
    uint16_t connectionTimeout;
    uint16_t dataTimeout;
} SCFG_Resp;

typedef struct SGACT_Resp
{
    PDP_CONTEXT_ID cid;
    bool active;
} SGACT_Resp;

/**
 * @brief The drivers of the GSM module Telit GE-910-Quad-V3 
 */
class Telit_GE910_Quad_V3
{

private:
    PlatformMutex mutex;
    SysLog *log;
    UARTSerial bus;
    ATCmdParser AT;
    DigitalOut shutdown;
    DigitalOut power_on_off;
    DigitalIn alarm;
    LowPowerTicker tasks;
    char buffer[AT_BUFFER_SIZE] = {'\0'};

    void SIM_not_inserted_ErrorHandler(void);
    void SIM_failure_ErrorHandler(void);
    void readLine();

    int selint();
    int regmode();
    int smsmode();

public:
    Telit_GE910_Quad_V3(SysLog *_log, const char *sim_pin);

    // Device ON/OFF
    void on(void);
    void off(void);
    bool isOn(void);
    void restart(void);

    // Error reporting
    int cmee(CMEE_Type cmee);
    CMEE_Type cmee(void);

    // Signal & network
    bool csq(CSQ_Resp *csq);
    bool cgreg(CGREG_Resp *cgreg);
    bool cgatt(CGATT_STATE config);
    CGATT_STATE cgatt();
    bool cgdcont(PDP_CONTEXT_ID cid, const char *pdp_type, const char *apn, const char *pdp_addr, bool data_compression, bool head_compression);
    bool cgdcont(CGDCONT_Resp _cgdcont[]);
    bool cgact(SGACT_Resp contexts[6]);
    bool cgact(PDP_CONTEXT_ID cid, bool activate, const char *username = 0, const char *password = 0);

    // Socket config
    bool scfg(SCFG_Resp config[6]);
    bool scfg(SocketIdentifier socketIdentifier, PDP_CONTEXT_ID cid, uint16_t packetSize = 300, uint16_t exchangeTimeout = 90, uint16_t connectionTimeout = 600, uint16_t dataTimeout = 50);

    // PIN & PUK management
    int cpin(const char *sim_pin);
    CPIN_Type cpin(void);
    int pct(void);

    // SMS services
    CMGF_Type cmgf(void);
    int cmgf(CMGF_Type cmgf);
    int cmgs(CMGS_Format_Type numberFormat, const char *phoneNumber, const char *message);

    // Internet services
    bool ping(PING_Resp *ping, const char *server, uint8_t retries = 4, uint16_t length = 32, uint16_t timeout = 50, uint8_t ttl = 128);
};`

**Comment by christophschlosser** [comment]
Thanks for sharing, I will take a look at it.
PS: I reformatted your reply to properly display

**Comment by samkearney** [comment]
I also have this issue if it helps.

**Comment by christophschlosser** [comment]
Thanks for your feedback. Right now I have a lot of other things to do and since it doesn't even affect all macros (and macros aren't really in the focus for documentation).

If you can help fix the bug that would be really helpful.

**Comment by neocliff** [comment]
I'm also having the same problem. My code looks like...

```C
#define VERSION_MAJOR 1
#define VERSION_MINOR 0
#define VERSION_PATCH 50
```

When I type "/**" followed by a RETURN I get...
```C
/**
 * @brief 
 * 
 */fine VERSION_MAJOR 1
#define VERSION_MINOR 0
#define VERSION_PATCH 50
```

If I can get some time I'll look at what might be causing the problem.

**Comment by christophschlosser** [comment]
Hi @klonq123 @neocliff  @samkearney ,

this will be fixed in 0.7.0 which I just triggered. The update should be visible in VS Code in ~15 minutes.
Hope you're still enjoying the extension and this makes your life a bit easier :)

**Comment by samkearney** [comment]
@christophschlosser I am and it does! Thank you!!
